### PR TITLE
Depend on Cargo and general cleanup

### LINKED ISSUE #118: error running `cargo apk`
I followed the tutorial in the README, but when i run `cargo apk` (note, no additional arguments) 
I get:
```
error: extra arguments to `rustc` can only be passed to one target, consider filtering
the package by passing e.g. `--lib` or `--bin NAME` to specify a single target
Error executing ["cargo", "rustc", "--target", "arm-linux-androideabi", "--", "-C", "linker=~/android-artifacts/linker_exe", "--extern", "cargo_apk_injected_glue=~/target/android-artifacts/arm-linux-androideabi/libcargo_apk_injected_glue.rlib"]
Status code 101
```

my toml
```toml
[package]
name = "rustdl"
version = "0.1.0"
[[bin]]
name = "rustd"
path = "src/main.rs"
[package.metadata.android]
label = "Basic android-rs-glue example"
[dependencies.android_glue]
version = "^0.2"
```

my hello world
```rust
extern crate android_glue;
fn main() { println!("hello world"); }
```

More information
```
NDK_HOME=android-ndk-r13b
```

Can i haz help plz?

### LINKED ISSUE #125: Cargo workspace support
With workspaces, there is a single `target` folder for the artefacts, but cargo-apk keeps creating one locally instead.

### PR DESCRIPTION
**Pull request by tomaka**
Fix #125 
Fix #118

### CODE REVIEW

File: cargo-apk/Cargo.toml [hunk 1]
```diff
@@ -1,12 +1,13 @@
 [package]
 name = "cargo-apk"
-version = "0.1.8"
+version = "0.1.7"
```
Comment (ozkriff) [inline cargo-apk/Cargo.toml]: `0.1.9`?

Comment (tomaka) [inline cargo-apk/Cargo.toml]: Err... For some reason I saw in the changelog that I went from 0.1.7 to 0.1.8, so I decided to revert to 0.1.7. I don't know what happened.

**Comment by jonas-schievink** [comment]
The commit email might be wrong, GitHub can't associate the commit with your account

**Comment by tomaka** [comment]
Note that this is a work-in-progress branch. I'm add commits one by one so that travis can check whether it is ok. My local connection is not good enough to bother downloading the SDK and NDK in order to put them in the Docker container.

Basically the plan of this PR is:

- Use clap to parse the command line arguments.
- Use cargo as a library in order to build the various elements.
- Add support for --bin, --lib, --example (#118).
- Put the artifacts in the correct directory (#125).

**Comment by tomaka** [comment]
Why must this be so freaking complicated.
I used docopt so that I could just copy-paste Cargo's USAGE, but this stupid program returns an error if you don't pass the exact program name that was specified.
If I use clap I have to rewrite all the arguments manually, but that's what I'm going to have to do.

**Comment by tomaka** [comment]
The next step of this PR would be to use the `cargo::compile` function to compile the crate instead of using `Command::new("cargo").arg("rustc")`, so that we can properly pass the build configuration instead of relying on hacks.

Unfortunately `cargo::compile` doesn't support passing environment variables, which is what the hacky linker currently uses. Therefore I'd have to rework the hacky linker to use command line arguments, but unfortunately I risk introducing bugs by doing so.

**Comment by tomaka** [comment]
Ok this PR is finally complete.
You can now finally use `cargo apk <same options as regular Cargo>`. The `target` directory will also be the same as Cargo.
You can also do `cargo apk install <same options as regular Cargo>`.

Note that for some reason `cargo apk --release` generates an APK that can't be installed.
But since this crate also needs to be updated for the latest versions of the SDK and the switch from ant to gradle, this should be fixed later.

**Comment by ozkriff** [comment]
> Note that for some reason cargo apk --release generates an APK that can't be installed.

I think it needs to be signed: https://stackoverflow.com/questions/10930331/how-to-sign-an-already-compiled-apk

**Pull request merged**